RadixWallet/Clients/OnLedgerEntitiesClient/OnLedgerEntitiesClient+Live.swift

```diff
@@ -251,12 +251,19 @@ extension OnLedgerEntitiesClient {
 					ledgerState: response.ledgerState
 				)
 
+				var allMetadataItems = Set(item.metadata.items)
+				if let nextCursor = item.metadata.nextCursor {
+					// Only fetch metadata if there is more after the first page returned.
+					let remaining = try await gatewayAPIClient.fetchEntityMetadata(item.address, ledgerState: response.ledgerState, nextCursor: nextCursor)
+					allMetadataItems.append(contentsOf: remaining)
+				}
```

Comment:
Not 100% sure about this, this will increase the overall loading times of the Wallet. Ideally the metadata should fetched only when going in the Resource Details.

Reply:
Actually it wasn't increasing the overall loading times, since we only fetch the actual resource when we try to open its details. Before that, we are actually requesting `state/entity/details` for the accounts (not the resources), and in such case we don't have any `item.metadata.nextCursor` (as it only fetches the explicit metadata of the resources inside the accounts).

However, I pushed a new commit that explicitly requests the metadata only when requested. It adds some noise, so let me know if you prefer which version you prefer.

Reply:
Almost true. We are fetching the resource details for the ones that are either part of staking or pool units, see [fetchPoolAndStakeUnitsDetails](https://github.com/radixdlt/babylon-wallet-ios/blob/5a676d23184651706492223f4080a9450738103b/RadixWallet/Clients/AccountPortfoliosClient/AccountPortfoliosClient%2BLive.swift#L127) or when categorising resources to pool units, stake units, nft claims